cpp/tests/copying/shift_tests.cu

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <limits>
+#include <memory>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/cudf_gtest.hpp>
+#include <tests/utilities/table_utilities.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/scalar/scalar.hpp>
+
+using cudf::test::fixed_width_column_wrapper;
+using TestTypes = cudf::test::Types<int32_t>;
+
```

Comment:
I don't think all the conditions that throw exceptions are exercised by these tests. Perhaps add a couple more simple tests to exercise them.